pkg/apis/servicecatalog/types.go

```diff
@@ -191,7 +191,7 @@ type ServiceClass struct {
 
 	// Plans is the list of ServicePlans for this ServiceClass.  All
 	// ServiceClasses have at least one ServicePlan.
-	Plans []ServicePlan
+	Plans []Plan
```

Comment:
If we're just splitting, plans should just reference the service class and we should remove this array field of `ServiceClass`.

Two-way references are very difficult to maintain.